src/app/campaign/store/reducers/campaign.reducer.ts

```diff
@@ -0,0 +1,103 @@
+import { CampaignActions } from '../actions/';
+import { CampaignActionTypes } from '../actions/campaign.action.types';
+import { HalDoc } from 'ngx-prx-styleguide';
+
```

Comment:
TODO: have a few too many Campaign interfaces declared in this app. There is still one in core/campaign.models because it's used by the campaign.service, but that campaign.service (and the rest of those core services) are only used by this campaign module (I think.) Probably will move them to the feature module where it makes sense to keep things more localized.

Reply:
We also talked about maybe just sending the HalDoc (or basic objects?) all the way out to the forms and letting them deal directly with the JSON our API returns.  Not sure I know enough to know why that's a bad idea.

Anyways ... not in this PR for sure, but might think about going forward.

Reply:
I do just about have a change ready that has the reducers converting the HalDoc to a plain object Campaign model instead of back in the services.